nemo/collections/tts/models/talknet.py

```diff
@@ -370,23 +373,27 @@ def list_available_models(cls) -> 'List[PretrainedModelInfo]':
         Returns:
             List of available pre-trained models.
         """
-        # list_of_models = []
-        # model = PretrainedModelInfo(
-        #     pretrained_model_name="tts_en_talknet",
-        #     location=(
-        #         "https://api.ngc.nvidia.com/v2/models/nvidia/nemo/tts_en_talknet/versions/1.0.0rc1/files"
-        #         "/talknet_spect.nemo"
-        #     ),
-        #     description=(
-        #         "This model is trained on LJSpeech sampled at 22050Hz, and can be used to generate female "
-        #         "English voices with an American accent."
-        #     ),
-        #     class_=cls,  # noqa
-        #     aliases=["TalkNet-22050Hz"],
-        # )
-        # list_of_models.append(model)
-        # return list_of_models
-
-        # NOTE: TalkNet loading is currently broken in main and newer. Please revert to r1.2.0 if interested in
-        # TalkNet.
-        pass
+        list_of_models = []
+        model = PretrainedModelInfo(
+            pretrained_model_name="tts_en_talknet",
+            location=(
+                "https://api.ngc.nvidia.com/v2/models/nvidia/nemo/tts_en_talknet/versions/1.0.0rc1/files"
+                "/talknet_spect.nemo"
+            ),
+            description=(
+                "This model is trained on LJSpeech sampled at 22050Hz, and can be used to generate female "
+                "English voices with an American accent."
+            ),
+            class_=cls,  # noqa
+            aliases=["TalkNet-22050Hz"],
+        )
+        list_of_models.append(model)
+        return list_of_models
+
+    @staticmethod
+    def from_pretrained_all(model_name: str, *args, **kwargs):
+        """Custom TalkNet's three-part load logic."""
+        model = TalkNetSpectModel.from_pretrained(model_name, *args, **kwargs)
+        model.add_module('_pitch_model', TalkNetPitchModel.from_pretrained(model_name, *args, **kwargs))
+        model.add_module('_durs_model', TalkNetDursModel.from_pretrained(model_name, *args, **kwargs))
+        return model
```

Comment:
Just redefine from_pretrained in your class, don't make a new function.

Reply:
That what I have in mind in the first place, but I just can't do this by design.
Since the method is static, we need `TalkNetSpectModel.from_pretrained` to be available during loading, otherwise it's gonna be an infinite cycle of calling from_pretrained func.

Reply:
- call `super().from_pretrained` instead of `TalkNetSpectModel.from_pretrained()`?
- Make a new model? `TalkNetFullModel` that defines all three as submodules and use that model for inference?
- Why is it a staticmethod? You have it defined as a classmethod and the base class's from_pretrained is also defined as a classmethod.

Reply:
great, thx
I am actually didn't know super() works for staticmethods